### DISCUSSION TITLE
[Feature request] Ability to enforce rules relative imports and absolute imports

### DISCUSSION THREAD

**Discussion by rishitank** [Igorkowalski94/eslint-plugin-project-structure#17] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently when working within a family of folders, or a private folder, we have a requirement to use relative imports. And anything imported outside of this family would use absolute imports. Is there a feature that supports this? Could it be something that would work with the independent module rules? 

**Describe the solution you'd like**
Ability to define whether imports should be relative or absolute (with support for path aliases) when imported from specified files that match the given pattern. For example:

```
src
  features
    feature1
      components
        component1
          component1.jsx
        component2
          component2.jsx
      index.js
    feature2
      components
        component1
          component1.jsx
        component2
          component2.jsx
      index.js
```

Files within feature 1 or files within feature 2 should use relative imports. However if a file from feature 1 imported a file from feature2 it should use an absolute import. This is just a general example, but it could be applicable to other scenarios such as pages accessing global components vs child components of a given page using relative imports. 

**Describe alternatives you've considered**
I have yet to find an eslint plugin that would allow me full control to do this. 

**Additional context**

**Comment by Igorkowalski94** [comment]
Overall, it's quite an unusual approach, teams usually choose either relative imports or non-relative imports exclusively. 

I need to think it over. If other community members show strong interest in this idea, I could consider creating such a rule.

**Comment by jarzebowsky** [comment]
The vote from my side comes. What we are doing is defining a families and within familiy (if possible) we do imports thru relative paths, if we import from outside (if possible) we use alias imports. On top of that we are also having the sort import order which quickly shows us what we import from where and how much. If that could be merged I would be happy to prepare such PR.
